src/app/groups/containers/user-group-invitations/user-group-invitations.component.ts

```diff
@@ -1,97 +1,116 @@
-        this.state = state.tag;
-        if (state.isReady) {
-          this.requests = state.data;
-        }
-      },
-      error: _err => {
-        this.state = 'fetchingError';
-      }
-    });
+    private processGroupInvitationService: ProcessGroupInvitationService,
+  ) {}
+
+  ngOnDestroy(): void {
+    this.sortSubject.complete();
   }
 
-  ngOnInit(): void {
-    this.dataFetching.next({ sort: this.currentSort });
+  onFetch(sort: string[]): void {
+    if (JSON.stringify(sort) !== JSON.stringify(this.sortSubject.value)) {
+      this.sortSubject.next(sort);
+    }
   }
 
-  ngOnDestroy(): void {
-    this.dataFetching.complete();
+  onCustomSort(event: SortEvent): void {
+    const sortMeta = event.multiSortMeta?.map(meta => (meta.order === -1 ? `-${meta.field}` : meta.field));
+    if (sortMeta) this.onFetch(sortMeta);
   }
 
-  onProcessRequests(params: { data: PendingRequest[], type: Action }): void {
-    this.state = 'processing';
-    this.requestActionService.processGroupInvitations(params.data.map(r => r.group.id), params.type)
-      .subscribe({
-        next: result => {
-          this.state = 'ready';
-          displayResponseToast(this.actionFeedbackService, parseGroupInvitationResults(result), params.type);
-          this.dataFetching.next({ sort: this.currentSort });
-          this.currentContentService.forceNavMenuReload();
-        },
-        error: err => {
-          this.state = 'ready';
-          this.actionFeedbackService.unexpectedError();
-          if (!(err instanceof HttpErrorResponse)) throw err;
+  onAccept(pendingRequest: PendingRequest): void {
+    this.processing = true;
+    this.processGroupInvitationService.accept(pendingRequest.group.id).subscribe({
+      next: result => {
+        this.processing = false;
+        if (!result.changed) {
+          this.actionFeedbackService.error(`Unable to accept invitation to group "${ pendingRequest.group.name }"`);
+          return;
         }
-      });
+        this.actionFeedbackService.success(`The ${ pendingRequest.group.name } group has been accepted`);
+        this.sortSubject.next(this.sortSubject.value);
+        this.currentContentService.forceNavMenuReload();
+      },
+      error: err => {
+        this.processing = false;
+        this.actionFeedbackService.unexpectedError();
+        if (!(err instanceof HttpErrorResponse)) throw err;
+      },
+    });
   }
 
-  onFetch(sort: string[]): void {
-    if (JSON.stringify(sort) !== JSON.stringify(this.currentSort)) {
-      this.currentSort = sort;
-      this.dataFetching.next({ sort: this.currentSort });
-    }
+  onReject(pendingRequest: PendingRequest): void {
+    this.processing = true;
+    this.processGroupInvitationService.reject(pendingRequest.group.id).subscribe({
+      next: result => {
+        this.processing = false;
+        if (!result.changed) {
+          this.actionFeedbackService.error(`Unable to reject invitation to group "${ pendingRequest.group.name }"`);
```

Comment:
Strings need to be localized. Check everywhere in this file.

Reply:
Fixed